tests.py

```diff
@@ -28,37 +31,39 @@ def run_vsm(self, name, input_data, expected_output, use_initial=True):
         process = Popen(cmd, stdin=PIPE, stdout=PIPE)
 
         output, _ = process.communicate(input=input_data.encode(), timeout=2)
+        output = output.decode()
+        print("result: {}".format(output))
+        self.assertEqual(output, expected_output)
 
-        self.assertEqual(output.decode() , expected_output)
 
     def test_simple0(self):
         input_data = 'transmission_gear = "reverse"'
-        expected_output = 'car.backup = True\n'
+        expected_output = 'transmission_gear = reverse\ncar.backup = True\n'
```

Comment:
When you show what is state with the State = {} marking (see below) you'll probably need a more readable way of assigning this string. Here's one way of doing it:

```
        expected_output = ('State = {\n'
                           '  transmission_gear = reverse\n'
                           '}\n'
                           'car.backup = True\n')
```